c++/include/orc/Common.hh

```diff
@@ -122,6 +122,11 @@ namespace orc {
     StreamKind_BLOOM_FILTER_UTF8 = 8
   };
 
+  enum ArrayReadIntent {
```

Comment:
Could you add some description for this new enum?

Reply:
Yes, we can extend this enum to support MAP and UNION type. However, I'd like to focus on LIST first in this PR, if it is OK. I'd be happy to extend it in the future JIRA.

Renaming this enum back to 'ReadIntent' sounds better if we anticipate extending it for MAP and UNION.

Reply:
Renamed back to `ReadIntent`, gave description comments, and TODO for MAP and UNION type.